paddle/majel/allocation.cc

```diff
@@ -0,0 +1,95 @@
+#include <boost/variant.hpp>
+
+#include <majel/allocation.h>
+
+namespace majel {
+namespace detail {
+
+class Allocator : public boost::static_visitor<void*> {
+public:
+  Allocator(size_t size) : size_(size) {}
+
+  void* operator()(const CpuPlace& p) const {
+    void* address;
+    CHECK_EQ(posix_memalign(&address, 32ul, size_), 0);
```

Comment:
Yes, I agree with @wangkuiyi . I think it's better to make majel an independent tensor library.